packages/ui/src/hooks/use-api.ts

```diff
@@ -0,0 +1,37 @@
+import { useState } from 'react';
+import { HTTPError } from 'ky';
+import { RequestErrorBody } from '@logto/schemas';
+
+function useApi<T extends any[], U>(api: (...arguments_: T) => Promise<U>) {
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState<RequestErrorBody | null>(null);
+  const [result, setResult] = useState<U>();
+
+  const fetch = async (...arguments_: T) => {
```

Comment:
the `_` looks ugly (i know because of the lint rule), i'll change it soon.

Reply:
wait for 15min i'll publish a new version of config

Reply:
it should be `run` or something instead of `fetch`? since we'll cover all the api methods

Reply:
under `packages/ui` run the command:

```
pnpm upgrade @logto/{eslint-config,eslint-config-react,ts-config,ts-config-react}
```

`0.1.0-rc.16` enabled curly rule and allows `args`